bundle/bundle.go

```diff
@@ -120,10 +120,17 @@ func NewFile(name, hash, alg string) FileInfo {
 // Manifest represents the manifest from a bundle. The manifest may contain
 // metadata such as the bundle revision.
 type Manifest struct {
-	Revision      string                 `json:"revision"`
-	Roots         *[]string              `json:"roots,omitempty"`
-	WasmResolvers []WasmResolver         `json:"wasm,omitempty"`
-	Metadata      map[string]interface{} `json:"metadata,omitempty"`
+	Revision      string         `json:"revision"`
+	Roots         *[]string      `json:"roots,omitempty"`
+	WasmResolvers []WasmResolver `json:"wasm,omitempty"`
+	// RegoVersion
+	// We don't use ast.RegoVersion here, as this iota type's order isn't guaranteed to be stable over time.
+	// We use a pointer so that we can support hand-made bundles that don't have an explicit version appropriately.
+	// E.g. in OPA 0.x if --v1-compatible is used when consuming the bundle, and there is no specified version,
+	// we should default to v1; if --v1-compatible isn't used, we should default to v0. In OPA 1.0, no --x-compatible
+	// flag and no explicit bundle version should default to v1.
+	RegoVersion *int                   `json:"rego_version,omitempty"`
```

Comment:
What you this value be set to if the bundle contains a mix of policies 🤔  For example this would be the case for a hand-made bundle. In that case would we need to list of the modules and their versions?

Reply:
We're operating under the assumption that all modules in a bundle has the same Rego version. Hand-made modules will need to do the same (the attr isn't mandatory, though). Keeping track of Rego versions for individual modules would be possible, but I think that could easily get messy. Even without such support, I think there is a path forward for mixed setups, as you can create separate bundles for your different Rego versions.

Reply:
The PR has been amended with per-file rego-version overrides.